packages/scanner/src/rules/jwtUnverifiedSignature.ts

```diff
@@ -0,0 +1,49 @@
+import { MatchResult, Rule, RuleLogic } from '../types';
+import parseRuleDescription from './lib/parseRuleDescription';
+import { Event } from '@appland/models';
+
+class JwtUnverifiedSignatureLogic implements RuleLogic {
+  matcher(event: Event): MatchResult[] | undefined {
+    let verified = false;
+    const matches = new Array<MatchResult>();
+
+    const children = [...event.descendants()];
+    while (children.length) {
+      const child = children.pop() as Event;
+      if (child.labels.has('jwt.signature.verify')) {
+        verified = true;
+        break;
+      }
+      children.push(...child.descendants());
```

Comment:
Descendants of the child should already be in the array you’ve build. Also you can avoid all this array building pushing and popping by using event navigator.